app/src/assets/localization/en/device_settings.json

```
@@ -226,7 +226,7 @@
   "resets_cannot_be_undone": "Resets cannot be undone",
   "restart_now": "Restart now?",
   "restart_robot_confirmation_description": "<span>It will take a few minutes for <bold>{{robotName}}</bold> to restart.</span>",
-  "restart_taking_too_long": "{{robotName}} restart is taking longer than expected. If the robot appears unresponsive, try restarting the robot manually.",
+  "restart_taking_too_long": "{{robotName}} restart is taking longer than expected. Check your robot's settings page to verify whether or not the update was successful. If the robot appears unresponsive, try restarting the robot manually.",
```

@ecormany Totally open for any ideas on how to phrase this!

---

Hmm. The robot settings page is already visible behind the modal. What exactly is the next action we want users to try in this scenario?

We should also keep in mind that we have a warning in our latest release notes that restarts may take a while due to database migration from 6.x -> 7.x. This string should be as general as possible — it should apply to all future versions — without contradicting this special case that we know a lot of users will have to go through soon.

---

So this text can also appear from the Devices page by clicking an update banner - a user can begin the robot update flow without necessarily being on the settings page. We'd ideally like users to check the update page to see if the update was applied (sometimes a robot won't properly tell the app that the update applied successfully, and we have to handle those cases). However, if the robot is offline, there's no way to tell the update went through without first restarting the robot... at this point, a manual restart would likely be required. 

And yes, completely agree about the migration notice. This string is meant to be generic and covers cases when the robot restarts and is unable to connect (wifi issues, hardware issues, etc etc).

---

OK this is pure suggestion, or you can take Shlok's GTG and merge as is 😁 
```suggestion
  "restart_taking_too_long": "{{robotName}} is taking longer than expected to restart. Check the Advanced tab of its settings page to see whether it finished updating. If the robot is unresponsive, restart it manually.",
```

---

I like it, thanks! 📝